.github/workflows/basic-continuous-integration.yml

```diff
@@ -7,22 +7,26 @@ jobs:
     name: PHPUnit Unit Tests
     runs-on: ubuntu-22.04
 
+    strategy:
+      matrix:
+        php: ['8.0', '8.1', '8.2', '8.3']
+
     steps:
       - name: Checkout Code
         uses: actions/checkout@v2
 
       - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
-          php-version: '8.0'
+          php-version: ${{ matrix.php }}
           ini-values: assert.exception=1, zend.assertions=1
 
       - name: Cache Composer packages
         uses: actions/cache@v2
         with:
-          path: ~/.composer/cache/files
-          key: composer-${{ hashFiles('**/composer.lock') }}
-          restore-keys: composer-
+          path: vendor
```

Comment:
One issue with caching the `vendor` folder is that composer's post-install scripts don't run. See https://github.com/actions/cache/blob/main/examples.md#php---composer for the recommended cache setup for composer.

Reply:
![image](https://github.com/user-attachments/assets/26a17658-52f9-4ee2-9493-5bbcc7607678)

Reply:
I'm not sure what the screenshot is saying. Caching the `vendor/` folder will technically work, but the recommendation is to cache the `composer config cache-files-dir` directory.

Reply:
I made it in the same way as on retain.
It works fine there.
Can change on the recommended value.

Reply:
Yeah, we should probably use the recommended way. I'll create a ticket to update Retain as well. Thanks!